network-structure.md

```diff
@@ -0,0 +1,29 @@
+# Network structure
+
+## Node
+A Node is a local context in which (part of) an ECP deployment runs. 
+This may be a single application using one or more threads or processes. 
+An ECP network has one or more Nodes. 
+If it has a single Node, its Components may use the Local Message Transport (LMT/"local mode"), see below. 
+If it has multiple Nodes, they must use the Distributed Message Transport (DMT/"distributed mode").
+
+(TBC) Optional feature "Bridging Coordinator": In a DMT network, if the Components of a Node use local mode, only a Coordinator in that Node may use DMT to bridge messages to/from other Nodes. Put differently, Components in local mode may only communicate to outside their Node via a Coordinator in their Node.
+:::{admonition} TODO
+Decide if we include the Bridging Coordinator concept (I think we will).
+:::
+:::{admonition} TODO
+Decide if the "mode" is the same for all Components in a Node, and if yes, how we track which Node a Component belongs to, and how the mode information propagates within.
```

Comment:
I'd say, that you can have both modes in the same node: If you tie Components to the local Coordinator, they use the local mode, or they can participate in the DMT (and talk to other Components in their node via the DMT and through the local Coordinator from the "outside").

One "implementation" idea, could be, that you give a socket to a Component, which can either be a zmq socket to a DMT Coordinator or some object connected (maybe using pipes, queues, but with the same methods as a zmq socket) to the local Coordinator.

Basically you give a different "adapter" (in pymeasure lingo).

Reply:
Hm, maybe we need to "pivot" this "mode" concept a little?
What I want to express/enable is that we have a couple of routing options:
* Use the Node-local Coordinator to route messages anywhere in the network (DMT, has to use zeromq tcpip)
* Use the Node-local Coordinator to route messages anywhere in the local Node (DMT, has to use zeromq tcpip)
* If you want (e.g. for increased reliability, performance, minimized network/port use...), you may: Use the Node-local Coordinator to route messages anywhere in the local Node (LMT, use zeromq inproc, queues or one of the other local options)

I would say that a Coordinator is exceptional, as it always needs one DMT connection (the one to the other Coordinators). The question is for the Node-facing "side" of the Coordinator (which can either be DMT or LMT): Should it be allowed to mix-and-match between LMT and DMT connections to its Components? Does allowing mixed use make the spec/logic much more complicated (for not much gain)? 
If it does not restrict us too much, I'd suggest to disallow mixing -- we can always relax that later, while it will be more difficult to start with mixed use and restrict it later.

Reply:
If you have another dmt Coordinator running, I do not see the benefit, that the node's local coordinator offers dmt connection to its components.

But it should not be too complex: The Coordinator polls all connections for incoming messages and handles them.
If there is one socket more or less, does not matter as long as the messages have the same format.

Reply:
Let us restrict it (hinting at the possibility to open it later), then we try to mix it experimentally. According to the results, we can allow a mix or not.

Reply:
> If you have another dmt Coordinator running, I do not see the benefit, that the node's local coordinator offers dmt connection to its components.

I made this point elsewhere already I think, but one advantage to use the Node-local Coordinator is closeness -- that "other DMT Coordinator" could be on another machine, which is slower and less reliable. 
_Should I add a note on this in the document?_
We could even specify that Components need to use their own Node's Coordinator.

Reply:
OK, I tightened up usage a bit (LMT/DMT choice per Node, exactly one Coordinator per Node). We can always relax later should we need it, but I want to try to keep complexity down so we don't get lost in details.